test/jest/tsconfig.json

```diff
@@ -2,5 +2,6 @@
   "compilerOptions": {
     "skipLibCheck": true,
     "jsx": "react",
+    "esModuleInterop": true
```

Comment:
If this is not the cause of downstream failure, let's keep it strict, so it can help checking types from SDK

Reply:
This wasn't causing any failures, just a series of warnings:

```bash
2022-08-16 17:27:32 + yarn test:unit --ci
2022-08-16 17:27:32 yarn run v1.21.1
2022-08-16 17:27:32 $ jest --ci
2022-08-16 17:27:34 ts-jest[config] (WARN) message TS151001: If you have issues related to imports, you should consider setting `esModuleInterop` to `true` in your TypeScript configuration file (usually `tsconfig.json`). See https://blogs.msdn.microsoft.com/typescript/2018/01/31/announcing-typescript-2-7/#easier-ecmascript-module-interoperability for more information.
2022-08-16 17:27:34 ts-jest[config] (WARN) message TS151001: If you have issues related to imports, you should consider setting `esModuleInterop` to `true` in your TypeScript configuration file (usually `tsconfig.json`). See https://blogs.msdn.microsoft.com/typescript/2018/01/31/announcing-typescript-2-7/#easier-ecmascript-module-interoperability for more information.
2022-08-16 17:27:34 ts-jest[config] (WARN) message TS151001: If you have issues related to imports, you should consider setting `esModuleInterop` to `true` in your TypeScript configuration file (usually `tsconfig.json`). See https://blogs.msdn.microsoft.com/typescript/2018/01/31/announcing-typescript-2-7/#easier-ecmascript-module-interoperability for more information.
2022-08-16 17:27:34 ts-jest[config] (WARN) message TS151001: If you have issues related to imports, you should consider setting `esModuleInterop` to `true` in your TypeScript configuration file (usually `tsconfig.json`). See https://blogs.msdn.microsoft.com/typescript/2018/01/31/announcing-typescript-2-7/#easier-ecmascript-module-interoperability for more information.
2022-08-16 17:27:34 ts-jest[config] (WARN) message TS151001: If you have issues related to imports, you should consider setting `esModuleInterop` to `true` in your TypeScript configuration file (usually `tsconfig.json`). See https://blogs.msdn.microsoft.com/typescript/2018/01/31/announcing-typescript-2-7/#easier-ecmascript-module-interoperability for more information.
...
```

Reply:
Tests still throws type errors/warnings
```bash
jared:react$ yarn test:unit
yarn run v1.22.17
$ jest
ts-jest[ts-compiler] (WARN) test/jest/oktaError.test.tsx:41:85 - error TS2322: Type 'string[]' is not assignable to type 'FieldError[]'.
  Type 'string' is not assignable to type 'FieldError'.

41       const error = new AuthApiError({ errorSummary, errorCode, errorLink, errorId, errorCauses });
                                                                                       ~~~~~~~~~~~

  ../../auth-js/build/lib/types/api.d.ts:66:5
    66     errorCauses?: Array<FieldError>;
           ~~~~~~~~~~~
    The expected type comes from property 'errorCauses' which is declared here on type 'APIError'
ts-jest[ts-compiler] (WARN) test/jest/reactRouterV6.test.tsx:42:20 - error TS2339: Property 'error' does not exist on type 'Readonly<{}>'.

42     if (this.state.error) {
                      ~~~~~
test/jest/reactRouterV6.test.tsx:44:30 - error TS2339: Property 'error' does not exist on type 'Readonly<{}>'.

44       return <p>{ this.state.error.toString() }</p>;
                                ~~~~~
 PASS  test/jest/oktaError.test.tsx (6.104 s)
 PASS  test/jest/reactRouterV6.test.tsx (6.75 s)
 PASS  test/jest/security.test.tsx (6.864 s)
 PASS  test/jest/secureRoute.test.tsx (7.068 s)
 PASS  test/jest/loginCallback.test.tsx (7.246 s)
 ```

Reply:
If tests still pass, we can revisit these warnings in a separate PR, fix types in tests might be a better option than setting loose ts compile configs.

Reply:
Removing this esInterop actually does cause tests to fail, I was mistaken